standalone-server/src/main/java/org/apache/calcite/avatica/standalone/StandaloneServer.java

```diff
@@ -92,7 +96,12 @@ public void join() throws InterruptedException {
 
   public static void main(String[] args) {
     final StandaloneServer server = new StandaloneServer();
-    new JCommander(server, args);
+    JCommander jc = new JCommander(server, args);
+    if (server.help) {
+      jc.usage();
+      System.exit(1);
```

Comment:
Since this might be expected you can easily add: 
`@SuppressForbidden(reason = "Allowed to exit explicitly here")`